packages/realm-server/migrations/1728561353454_create-user-billing.js

```diff
@@ -0,0 +1,202 @@
+exports.up = (pgm) => {
+  pgm.createTable('users', {
+    id: {
+      type: 'uuid',
+      primaryKey: true,
+      default: pgm.func('gen_random_uuid()'),
+    },
+    matrix_user_id: { type: 'varchar', notNull: true, unique: true },
+    stripe_customer_id: { type: 'varchar' },
+    created_at: {
+      type: 'timestamp',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  });
+
+  pgm.createTable('plans', {
+    id: {
+      type: 'uuid',
+      primaryKey: true,
+      default: pgm.func('gen_random_uuid()'),
+    },
+    name: { type: 'varchar', notNull: true },
+    monthly_price: { type: 'numeric', notNull: true },
+    credits_included: { type: 'integer', notNull: true },
+    created_at: {
+      type: 'timestamp',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  });
+
+  pgm.createTable('subscription_cycles', {
+    id: {
+      type: 'uuid',
+      primaryKey: true,
+      default: pgm.func('gen_random_uuid()'),
+    },
+    subscription_id: { type: 'uuid' },
+    period_start: { type: 'timestamp', notNull: true },
+    period_end: { type: 'timestamp', notNull: true },
+  });
+
+  pgm.createType('subscription_status', ['active', 'canceled', 'expired']);
+
+  pgm.createTable('subscriptions', {
+    id: {
+      type: 'uuid',
+      primaryKey: true,
+      default: pgm.func('gen_random_uuid()'),
+    },
+    user_id: { type: 'uuid', references: 'users(id)' },
+    plan_id: { type: 'uuid', references: 'plans(id)' },
+    started_at: { type: 'timestamp', notNull: true },
+    ended_at: { type: 'timestamp', notNull: true },
+    status: { type: 'subscription_status', notNull: true },
+    stripe_subscription_id: { type: 'varchar', notNull: true },
+    current_billing_cycle_id: {
+      type: 'uuid',
+      references: 'subscription_cycles(id)',
+    },
+  });
+
+  pgm.addConstraint(
+    'subscription_cycles',
+    'subscription_cycles_subscription_id_fkey',
+    {
+      foreignKeys: {
+        columns: 'subscription_id',
+        references: 'subscriptions(id)',
+      },
+    },
+  );
+
+  pgm.createTable('ai_actions', {
+    id: {
+      type: 'uuid',
+      primaryKey: true,
+      default: pgm.func('gen_random_uuid()'),
+    },
+    user_id: { type: 'uuid', references: 'users(id)' },
+    cost_in_usd: { type: 'numeric', notNull: true },
+    created_at: {
+      type: 'timestamp',
+      notNull: true,
+      default: pgm.func('current_timestamp'),
+    },
+  });
+
+  pgm.createType('balance_change_type', [
+    'plan_allowance',
+    'extra_credit_balance',
+  ]);
+
+  pgm.createTable('credit_balance_changes', {
+    id: {
+      type: 'uuid',
+      primaryKey: true,
+      default: pgm.func('gen_random_uuid()'),
+    },
+    user_id: { type: 'uuid', references: 'users(id)' },
+    change_in_credits: { type: 'numeric', notNull: true }, // can be negative
+    balance_change_type: { type: 'balance_change_type', notNull: true },
```

Comment:
maybe a better name is `balance_event_type`

Reply:
Will adjust it.